pkg/apis/wmc/v1alpha1/windowsmachineconfig_types.go

```diff
@@ -41,11 +42,43 @@ type Azure struct {
 
 // WindowsMachineConfigStatus defines the observed state of WindowsMachineConfig
 type WindowsMachineConfigStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+
+	// ProvisionedVMCount is the number of running VMs provisioned by the operator
+	ProvisionedVMCount int `json:"createdVMCount"`
+	// JoinedVMCount is the number of VMs that successfully joined the cluster
+	JoinedVMCount int `json:"joinedVMCount"`
+	// Conditions states the latest available observations of current state
+	Conditions []WindowsMachineConfigCondition `json:"conditions"`
+}
+
+type WindowsMachineConfigCondition struct {
+	// Type describes the type of the condition
+	Type WindowsMachineConfigConditionType `json:"type"`
+	// Status of the condition
+	Status corev1.ConditionStatus `json:"status"`
+	// LastTransitionTime is the timestamp corresponding to the last status change of this condition
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+	// Reason is the machine readable reason for the condition change
+	Reason string `json:"reason"`
+	// Message is the human readable reason for the condition change
+	Message string `json:"message"`
 }
 
+type WindowsMachineConfigConditionType string
+
+const (
+	// CloudProviderAPIConnection indicates if we were able to use the cloud provider API with the provided credentials
+	CloudProviderAPIConnected WindowsMachineConfigConditionType = "CloudProviderAPIConnected"
+	// VMProvisionFailure indicates if there was an issue provisioning a new VM
+	VMProvisionFailure WindowsMachineConfigConditionType = "VMProvisionFailure"
```

Comment:
I think it's better to talk in terms of Windows node objects instead of VM terminology considering we might move to machine-api. How about `WindowsNodeBootstrapperConfiguration` error and `WindowsNodeNetworkConfiguration` error.

Reply:
`WindowsNodeBootstrapperConfiguration` and `WindowsNodeNetworkConfiguration` don't sound like errors. We should come up with these error conditions given where we stand today and not based on the something in the future. When that future is imminent we can change these conditions. 

I am not averse to making this more node related but what I would like is that they accurately reflect what the error is. If there is failure to even create the VM, then it should say so. If there is a failure after a node object is created, then it should reflect that and so on.

Reply:
> WindowsNodeBootstrapperConfiguration and WindowsNodeNetworkConfiguration don't sound like errors. We should come up with these error conditions given where we stand today and not based on the something in the future. When that future is imminent we can change these conditions.

I am fine as long as we're ready to change it.


>  If there is failure to even create the VM, then it should say so. If there is a failure after a node object is created, then it should reflect that and so on.

I'd prefer the condition to be present but the message can be different. For 

For example `WindowsNodeBootstrapperTypeError` can point to error being created by bootstrapper and in the reason or message we can point to exact reason. Please look at the [scheduler status](https://github.com/openshift/cluster-kube-scheduler-operator/blob/6304b4bce9cd076966fdca532de5e3c652190734/pkg/operator/target_config_reconciler_v410_00.go#L59)

Reply:
>I think it's better to talk in terms of Windows node objects

I've been thinking about removing the `ProvisionedVMCount ` status object. I think it doesn't add any valuable information that `JoinedVMCount` doesn't already contain. Its existence adds the additional requirement of us having to update the fields in real time (as in constantly during reconciliation instead of at the end of reconciliation) as well. If we didnt the two fields would always have the same value as we tear down VMs that fail to become nodes. Because of this I think removing `ProvisionedVMCount ` is the way to go.

>I think it's better to talk in terms of Windows node objects instead of VM terminology 

Since we are using WNI right now, I think `VMProvisionFailure` adds a lot of information. I'm inclined to keep it, and re-evaluate as our codebase changes